pkgs/dart_tooling_mcp_server/lib/src/mixins/analyzer.dart

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:dart_mcp/server.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+import 'package:watcher/watcher.dart';
+
+/// Mix this in to any MCPServer to add support for analyzing Dart projects.
+///
+/// The MCPServer must already have the [ToolsSupport] mixin applied.
+base mixin DartAnalyzerSupport on ToolsSupport, LoggingSupport {
+  /// The analyzed contexts.
+  AnalysisContextCollection? _analysisContexts;
+
+  /// All active directory watcher streams.
+  ///
+  /// The watcher package doesn't let you close watchers, you just have to
+  /// stop listening to their streams instead, so we store the stream
+  /// subscriptions instead of the watchers themselves.
+  final List<StreamSubscription> _watchSubscriptions = [];
+
+  @override
+  FutureOr<InitializeResult> initialize(InitializeRequest request) {
+    if (request.capabilities.roots == null) {
+      throw StateError(
+          'This server requires the "roots" capability to be implemented.');
+    }
+    registerTool(analyzeFilesTool, _analyzeFiles);
+    initialized.then(_listenForRoots);
+    return super.initialize(request);
+  }
+
+  @override
+  Future<void> shutdown() async {
+    await super.shutdown();
```

Comment:
should super.shutdown get called last like we do for the dispose lifecycle method in flutter? Or should this be the first line in the method?

Reply:
it doesn't matter, it just needs to get called